pandas/plotting/_core.py

```diff
@@ -861,11 +863,42 @@ def _plot_colorbar(self, ax, **kwds):
 class ScatterPlot(PlanePlot):
     _kind = 'scatter'
 
-    def __init__(self, data, x, y, s=None, c=None, **kwargs):
+    def __init__(self, data, x, y, s=None, c=None, size_factor=1, **kwargs):
         if s is None:
-            # hide the matplotlib default for size, in case we want to change
-            # the handling of this argument later
-            s = 20
+            # Set default size if no argument is given.
+            s = 20 * size_factor
+        elif is_hashable(s) and s in data.columns:
+            # Handle the case where s is a label of a column of the df.
+            # The data is normalized to 200 * size_factor.
+            size_data = data[s]
+            self.bubble_points = 200
+            if is_categorical_dtype(size_data):
+                if size_data.cat.ordered:
+                    self.size_title = s
+                    self.bubble_dtype = 'categorical'
+                    size_data_codes = size_data.cat.codes + 1
+                    self.labels = list(size_data.cat.categories)[::-1]
+                    n_categories = len(self.labels)
+                    self.legend_bubbles_sizes = \
+                        (np.array(range(n_categories)) + 1)**2 * \
+                        self.bubble_points * size_factor / \
+                        size_data_codes.max()**2
+                    self.legend_bubbles_sizes = self.legend_bubbles_sizes[::-1]
+                    s = self.bubble_points * size_factor * \
+                        size_data_codes**2 / size_data_codes.max()**2
+                else:
+                    raise TypeError(
+                        "'s' must be numeric or ordered categorical dtype")
+            elif is_numeric_dtype(size_data):
+                self.bubble_dtype = 'numeric'
+                self.size_title = s
+                self.s_data_max = size_data.max()
+                self.size_factor = size_factor
+                s = self.bubble_points * size_factor * size_data / \
```

Comment:
Style: we prefer parenthesis to break long lines.

Reply:
OK